clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -237,8 +237,11 @@ mlir::Type CIRGenTypes::convertRecordDeclType(const clang::RecordDecl *rd) {
   assert(insertResult && "isSafeToCovert() should have caught this.");
 
   // Force conversion of non-virtual base classes recursively.
-  if (isa<CXXRecordDecl>(rd)) {
-    cgm.errorNYI(rd->getSourceRange(), "CXXRecordDecl");
+  if (const auto *cxxRecordDecl = dyn_cast<CXXRecordDecl>(rd)) {
+    if (cxxRecordDecl->getNumBases() > 0) {
+      cgm.errorNYI(rd->getSourceRange(),
```

Comment:
Implementing this would just be a recursive call here, so we could add this right now. Or did you refrain from doing this because we can't write a test for this at the moment?

Reply:
Yes, I left it out because we haven't done anything for derived classes in general.